src/canvas/App.tsx

```diff
@@ -0,0 +1,18 @@
+import { Renderer } from './Renderer';
+import { Document } from './Document';
+import { PathSelection } from './PathSelection';
+
+class App {
+    renderer: Renderer;
+    document: Document;
+    pathSelection: PathSelection;
+
+    constructor () {
+        this.document = new Document();
```

Comment:
I see what you mean. However, I think Document is a pretty common name for similar models. So, I am ok with keeping Document as the name. I think Pattern is confusing because we may want to have multiple Documents that in the end create one pattern. If we wanted to change it we could go with something like PDocument.

Reply:
ok, I didn't know Document was standard naming.

Reply:
ok I'm good with document.